tests/suite/upstream_settings_test.go

```diff
@@ -0,0 +1,523 @@
+				baseTeaURL := fmt.Sprintf("http://cafe.example.com:%d%s", port, "/tea")
+
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseCoffeeURL, address, "URI: /coffee")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(1000 * time.Millisecond).
+					Should(Succeed())
+
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseTeaURL, address, "URI: /tea")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(1000 * time.Millisecond).
+					Should(Succeed())
+			})
+		})
+
+		Context("nginx directives", func() {
+			var conf *framework.Payload
+
+			BeforeAll(func() {
+				podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podNames).To(HaveLen(1))
+
+				ngfPodName := podNames[0]
+
+				conf, err = resourceManager.GetNginxConfig(ngfPodName, ngfNamespace)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			DescribeTable("are set properly for",
+				func(expCfgs []framework.ExpectedNginxField) {
+					for _, expCfg := range expCfgs {
+						Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+					}
+				},
+				Entry("Coffee upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "upstream",
+						Value:     "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "zone",
+						Value:     "uspolicy_coffee_80 512k",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive",
+						Value:     "100",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_requests",
+						Value:     "55",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_time",
+						Value:     "1m",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_timeout",
+						Value:     "5h",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+				}),
+				Entry("Tea upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "zone",
+						Value:     "uspolicy_tea_80 128k",
+						Upstream:  "uspolicy_tea_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "upstream",
+						Value:     "uspolicy_tea_80",
+						File:      "http.conf",
+					},
+				}),
+			)
+		})
+	})
+
+	When("UpstreamSettingsPolicy targets a Service that does not exists", func() {
+		Specify("upstreamSettingsPolicy sets no condition", func() {
+			files := []string{"upstream-settings-policy/invalid-svc-usps.yaml"}
+
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			uspolicyNsName := types.NamespacedName{Name: "usps-target-not-found", Namespace: namespace}
+			Consistently(
```

Comment:
Bit inconsistent to have some functions that call the wait.PollUntil function, and some are just wrapped by Consistently. Let's pick a strategy and stick with it for all waiting functions.

Reply:
Since we are using `wait.PollUntilContextCancelled` for all other tests. I have updated the consistent matcher with that too for consistency